pkg/resources/fluentd/outputsecret.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2019 Banzai Cloud
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fluentd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/banzaicloud/logging-operator/pkg/model/secret"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *Reconciler) outputSecret(secrets *secret.MountSecrets, mountPath string) runtime.Object {
```

Comment:
it would be more clear to return the source secret items from this function and reconcile them outside instead of doing it inline